pkg/frontend/querymiddleware/model.proto

```diff
@@ -100,4 +100,11 @@ message Options {
 message Hints {
   // Total number of queries that are expected to to be executed to serve the original request.
   int32 TotalQueries = 1;
+  // Estimated total number of series that a request might return.
+  uint64 EstimatedCardinality = 2;
 }
+
+// This is used to store cardinality information in a cache.
+message QueryCardinality {
```

Comment:
[design] This is very specific. Why not something like `QueryStatistics` where in the future we can add more stuff inside? Then we can rename `Estimated` to `FetchedSeries`. All in all what we're tracking are query statistics not an estimation. Based on the statistics we then compute an estimation.

Reply:
I called the field `EstimatedSeriesCount`. I feel `FetchedSeries` is a bit misleading because the estimate will differ from the actual count of fetched series.